modules/auth/user_form.go

```diff
@@ -93,19 +93,25 @@ type UpdateProfileForm struct {
 	Email    string `binding:"Required;Email;MaxSize(254)"`
 	Website  string `binding:"Url;MaxSize(100)"`
 	Location string `binding:"MaxSize(50)"`
-	Gravatar string `binding:"OmitEmpty;Email;MaxSize(254)"`
 }
 
 func (f *UpdateProfileForm) Validate(ctx *macaron.Context, errs binding.Errors) binding.Errors {
 	return validate(errs, ctx.Data, f, ctx.Locale)
 }
 
-type UploadAvatarForm struct {
-	Enable bool
+const (
+	AVATAR_LOCAL string = "local"
+	AVATAR_BYMAIL string = "bymail"
```

Comment:
On Mon, Jun 27, 2016 at 02:36:35AM -0700, 无闻 wrote:
> >  }
> >  
> >  func (f *UpdateProfileForm) Validate(ctx *macaron.Context, errs binding.Errors) binding.Errors {
> >  	return validate(errs, ctx.Data, f, ctx.Locale)
> >  }
> >  
> > -type UploadAvatarForm struct {
> > -	Enable bool
> > +const (
> > +	AVATAR_LOCAL string = "local"
> > +	AVATAR_BYMAIL string = "bymail"
> 
> Seems never been used?

routers/org/setting.go: form.Source = auth.AVATAR_LOCAL

AVATAR_BYMAIL is currently never used as !AVATAR_LOCAL
is currently enough to know it's AVATAR_BYMAIL instead,
being there only 2 options. In case this grows, there
could be more values (for example when looking up avatar
by URL instead, via OpenID).

--strk;